### PR TITLE
Fix local api setup and README instructions

### LINKED ISSUE #1338: [API] Fix local setup and README instructions
* [x] The newest version of vercel CLI creates a `.vercel` folder with a `.env.development.local` file
 - dotenv invocations need to be updated to reference this instead of the root `.env.local`
* [x] Save this path to a constant and import it in both locations as `LOCAL_ENV_PATH`
* [x] Update the readme to explain what the expected behavior should be using vercel link && pull
- `vercel link` should be run from the KlimaDAO root
- must be a member of the KlimaDAO project in vercel
- say `Y`es when it asks if we want to link to an existing project, type `carbonmark-api`
- note that vercel pull --environment development should create the file in .vercel folder as noted above

### PR DESCRIPTION
**Pull request by psparacino**
## Description

Updated readme for updating to the newest vercel version, linking to the carbonmark-api vercel project, pulling the env variables, upgrading from previous version and running carbonmark-api locally in dev mode.

Updated path to the folder and variables created by the newest vercel cli.

## Related Ticket

Resolves #1338 

## Checklist

- [X] I have run `npm run build-all` without errors
- [X] I have formatted JS and TS files with `npm run format-all`

### CODE REVIEW

File: carbonmark-api/src/app.ts [hunk 1]
```diff
@@ -2,9 +2,10 @@ import AutoLoad, { AutoloadPluginOptions } from "@fastify/autoload";
 import * as dotenv from "dotenv";
 import { FastifyPluginAsync } from "fastify";
 import { join } from "path";
+import { LOCAL_ENV_PATH } from "./utils/helpers/utils.constants";
 
 // This is not pretty but we need to reference the env file at the root of the monorepo
-const result = dotenv.config({ path: "../.env.local" });
+const result = dotenv.config({ path: LOCAL_ENV_PATH });
```
Comment (0xemc) [inline carbonmark-api/src/app.ts:8]: Nice one and thanks for the extra docs 👍

Could you use `process.cwd` here plz ([example](https://github.com/KlimaDAO/klimadao/pull/1347/files#diff-bc149aa836d2ff3c5bd38465633d65ab9944fa49efa7687204654e8ee247baf5R7))  and remove the constant from utils?

```suggestion
const result = dotenv.config({ path: join(process.cwd(), ".env.local") });
```

**Comment by psparacino** [comment]
@0xemc I saw your comment about changing to process.cwd in app.ts, but can't find it on here(?). Did you delete and is that change no longer needed?

File: carbonmark-api/README.md [hunk 2]
```diff
@@ -1,24 +1,37 @@
 <!-- PROJECT LOGO -->
+ver
```
Comment (Atmosfearful) [inline carbonmark-api/README.md:2]:
```suggestion
```

**Review by Atmosfearful — APPROVED** [review]

**Comment by 0xemc** [comment]
> @0xemc I saw your comment about changing to process.cwd in app.ts, but can't find it on here(?). Did you delete and is that change no longer needed?

Hey no problem for now, there were other issues in the serverless version so I reverted it

**Pull request merged**